tox.ini

```diff
@@ -1,15 +1,108 @@
+# Here is what you can do:
+
+# tox                    # run tests with pytest (you can pass extra arguments
+#                          for pytest, e.g., "tox -- --update-goldens")
+# tox -e regexlint       # lint regular expressions with regexlint
+# tox -e pylint          # lint code with pylint
+# tox -e check           # perform various checks, including linting with pyflakes
+# tox -e mapfiles        # regenerate map files
+# tox -e coverage        # run tests and output a coverage report
+# tox -e doc             # compile documentation (you can pass extra arguments for
+#                          for Sphinx, e.g., "tox -e doc -- -b dirhtml" to select the
+#                          dirhtml builder)
+# tox -e web-doc         # compile documentation with demo support (uses Pyodide to
+#                          build Pygments for WASM)
+
 [tox]
-envlist = py{37, 38, 39, 310, 311, 312}, lint
+envlist = py
 
 [testenv]
+description =
+  run tests with pytest (you can pass extra arguments for pytest,
+  e.g., "tox -- --update-goldens")
 deps =
-    pytest
+    pytest >= 7.0
     pytest-cov
+    pytest-randomly
     wcag-contrast-ratio
 commands = pytest {posargs}
 
-
-[testenv:lint]
+[testenv:regexlint]
+description =
+  lint regular expressions with regexlint
 deps =
     git+https://github.com/pygments/regexlint.git@master
 commands = regexlint pygments.lexers
+
+
+[testenv:pylint]
+description =
+  lint code with pylint
+deps =
+  pylint
+commands =
+  pylint --rcfile scripts/pylintrc pygments
+
+
+[testenv:check]
+description =
+  miscellaneous checks on the source code, including pyflakes
+deps =
+  flake8
+commands =
+  # FIXME: What is build/ ??
+  python scripts/check_crlf.py pygments build external
+  python scripts/detect_missing_analyse_text.py --skip-no-aliases
+  # We only use pyflakes, not pycodestyle, but use it through flake8 nevertheless
+  # to be able to use the --ignore option.
+  # TODO: ignore result like in Makefile?
+  flake8 --select F --ignore F401 pygments
```

Comment:
I've yet to fix all linter issues :(

Reply:
Looks like it. The `tests/examplefiles` directory should be excluded, and I generally try to clean up linter warnings when I go over a file. I'd keep it (and fix the remaining ... what, 8 issues?) That said, this doesn't seem quite right unless it's ignoring "line too long" because I've seen tons of pep8 violations. For example, E302 is violated all over the place (two blank lines before `def`) We could run [`black`](https://pypi.org/project/black/) to fix all formatting, and then work through the remaining issues found by `flake8`.

Reply:
I'm not a fan of a big reformatting commit either, but I do think we should strive to be "lint clean" for logic issues (I just looked at `count_token_references` and it's fairly trivial stuff), and eventually this should be part of what was `make check` to avoid people even committing new issues.

I'll continue to fix formatting issues as I come across them :)

Reply:
No black please - it makes the token definitions about 2-5x less readable, in my very scientific scale :smile: 

The PEP8 part of flake8 can be useful if heavily customized.